scripts/west_commands/run_common.py

```diff
@@ -421,7 +421,7 @@ def do_run_common_image(command, user_args, user_runner_args, used_cmds,
     # the board has enabled this functionality, check if the board should be
     # reset or not. If this is not specified in the board/soc file, leave it up to
     # the runner's default configuration to decide if a reset should occur.
-    if runner_cls.capabilities().reset and '--no-reset' not in final_argv:
+    if runner_cls.capabilities().reset and '--no-reset' not in final_argv and len(board_image_count):
```

Comment:
the comment should be updated about what is this doing, I don't understand

Reply:
This is not something that should be addressed in this PR. The intention of the comment makes sense, the implementation uses a default set of conditions for multiple images if none are passed in as parameters, the only update I made is it doesn't properly check or test rather this code runs if there's only one image so I added the check. I have nothing to do with the implementation so maybe my assumption of the comment is wrong as well thus I cannot make the change.

Reply:
Hopefully the updated commit messages provide a better explanation about this,
what I was trying to say is the comment is only wrong because it was not checking for all cases, I only updated the check to match what the comment says. Updating the comment though I don't think should be in this PR since I don't have a high level overview of the intended before and after state for this function so there is a chance that the if check is correct but we were never giving the correct values. My fix was just to match the check with whats stated above in the comment.